src/shutdown.cc

```diff
@@ -499,7 +535,16 @@ void do_system_shutdown(shutdown_type_t shutdown_type)
 #ifdef __NetBSD__
     reboot(reboot_type, NULL);
 #else
-    reboot(reboot_type);
+    if (reboot(reboot_type)) {
+        // we're in trouble now
+        sub_buf.append("reboot(2): ");
+        sub_buf.append(strerror(errno));
+        sub_buf.append(
+                "\nThis may happen if you try to kexec without loading an image first, or if\n"
```

Comment:
The message should be made specific to the shutdown type, i.e. "if you try to kexec without loading an image" message should only be shown if the shutdown type is KEXEC.

Also check tone. Eg: "It is possible that no suitable kernel image was loaded before shutdown (with kexec) was issued" is more suitable.